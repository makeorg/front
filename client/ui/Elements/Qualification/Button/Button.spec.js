import React from 'react';
import snapshotDiff from 'snapshot-diff';
import renderer from 'react-test-renderer';
import { QualificationButtonElement } from './index';

describe('QualificationButtonElement', () => {
  const defaultProps = {
    color: 'red',
    label: 'foo'
  };

  it('must match the snapshot with default Props', () => {
    const component = renderer.create(
      <QualificationButtonElement {...defaultProps} />
    );
    expect(component).toMatchSnapshot();
  });

  it('must return the diff between snapshot with a negative vs a postive Tab Index', () => {
    const NegativeTabIndex = renderer.create(
      <QualificationButtonElement {...defaultProps} tabIndex="-1" />
    );
    const PositiveTabIndex = renderer.create(
      <QualificationButtonElement {...defaultProps} tabIndex="0" />
    );
    expect(snapshotDiff(NegativeTabIndex, PositiveTabIndex)).toMatchSnapshot();
  });

  it('must return the diff between snapshot when proposal is qualified or not', () => {
    const ButtonIsQualified = renderer.create(
      <QualificationButtonElement {...defaultProps} isQualified />
    );
    const ButtonIsNotQualified = renderer.create(
      <QualificationButtonElement {...defaultProps} isQualified={false} />
    );
    expect(snapshotDiff(ButtonIsQualified, ButtonIsNotQualified)).toMatchSnapshot();
  });
});
