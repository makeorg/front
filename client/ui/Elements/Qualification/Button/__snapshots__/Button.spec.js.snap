// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QualificationButtonElement must match the snapshot with default Props 1`] = `
<ButtonStyle
  aria-busy={false}
  aria-label="foo"
  as="QualifyButtonStyle"
  color="red"
  data-cy-button="qualification"
  onClick={[Function]}
>
  <span
    aria-hidden={true}
  >
    foo
  </span>
  <CounterStyle
    aria-hidden={true}
  >
    +1
  </CounterStyle>
</ButtonStyle>
`;

exports[`QualificationButtonElement must return the diff between snapshot when proposal is qualified or not 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -1,9 +1,9 @@
  <ButtonStyle
    aria-busy={false}
    aria-label="foo"
-   as="IsQualifiedButtonStyle"
+   as="QualifyButtonStyle"
    color="red"
    data-cy-button="qualification"
    onClick={[Function onClick]}
  >
    <span
@@ -12,8 +12,8 @@
      foo
    </span>
    <CounterStyle
      aria-hidden={true}
    >
-     0
+     +1
    </CounterStyle>
  </ButtonStyle>
`;
